### PR TITLE
Tallies added for teams and loadouts

### PR DESCRIPTION
**Pull request by maoadrian12**
## Describe your changes

Added a tally when looking at databases for the number of teams and loadouts.

## Issue or discord link

https://github.com/frzyc/genshin-optimizer/issues/2040

## Testing/validation

BEFORE
image
AFTER
image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [ x ] I have commented my code in hard-to understand areas.
- [  ] I have made corresponding changes to README or wiki.
- [  ] For front-end changes, I have updated the corresponding English translations.
- [ x ] I have run `yarn run mini-ci` locally to validate format and lint.
- [ x ] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Review by frzyc — CHANGES_REQUESTED** [review]
image
Is it possible to move the last edit time here?, Will balance out the whitespace.
Also, update translations in https://github.com/frzyc/genshin-optimizer/blob/master/libs/gi/localization/assets/locales/en/settings.json

**Comment by maoadrian12** [comment]
Looks like this now:
image
Not sure if this is intended behavior or not, but when the loadouts are changed/updated, it doesn't change the lastEdit time listed under the database.

**Comment by frzyc** [comment]
> Looks like this now: image Not sure if this is intended behavior or not, but when the loadouts are changed/updated, it doesn't change the lastEdit time listed under the database.

Looks much better now. That is a good catch on the `lastEdit` issue: https://github.com/frzyc/genshin-optimizer/issues/2072

**Review by frzyc — CHANGES_REQUESTED** [review]
Need to pass format CI stage:
assuming you are running a fork with upstream set to the main repo, you can run `nx format write --base upstream/master`

File: libs/gi/ui/src/components/database/DatabaseCard.tsx [hunk 1]
```diff
@@ -63,7 +63,17 @@ function DataCard({ index, readOnly }: { index: number; readOnly: boolean }) {
   const numChar = database.chars.keys.length
   const numArt = database.arts.values.length
   const numWeapon = database.weapons.values.length
-  const hasData = Boolean(numChar || numArt || numWeapon)
+  const numTeams = database.teams.values.length
+  let numLoadouts = 0;
```
Comment (frzyc) [inline libs/gi/ui/src/components/database/DatabaseCard.tsx:67]: The number of loadouts should just be `database.teamChars.values.length`. 

Can you also include the number of builds?
`database.builds.values.length`

**Review by frzyc — CHANGES_REQUESTED** [review]

**Comment by maoadrian12** [comment]
> Need to pass format CI stage: assuming you are running a fork with upstream set to the main repo, you can run `nx format write --base upstream/master`. Next time, it might be better to create a feature branch in your fork to avoid this issue.

How should I know if the formatting is correct, when I run the command(yarn run nx format:check) I get a lot of CRLF warnings, is that fine?
Besides that, here's what the screen looks like now:
image

**Comment by frzyc** [comment]
> I get a lot of CRLF warnings, is that fine?

I think you need to change your autocrlf settings in git:
```
git config core.autocrlf false
```

**Comment by maoadrian12** [comment]
When I do yarn run mini-ci, I only fail the gi-dm-localization test because the file at libs/gi/dm/GenshinData/TextMap/TextMapCHS.json isn't found, is this fine?

**Comment by frzyc** [comment]
That missing file is likely due to the datamine submodule not being loaded, you can run `reload-dm` for that. For the format issue, you can follow the instructions here https://github.com/frzyc/genshin-optimizer/pull/2070#pullrequestreview-2039740685. (mini-ci is suppose to run format, but it won't work for you here because you are working off your fork's master)

**Comment by maoadrian12** [comment]
Did I break something? The checks have been going on for a while

**Comment by frzyc** [comment]
> Did I break something? The checks have been going on for a while

Nope, because you are a first contributor, the pipeline needs approval before running. I've started it.

**Review by frzyc — APPROVED** [review]

**Comment by frzyc** [comment]
LGTM, thanks for your work!.

**Pull request merged**